core/os/os.cpp

```diff
@@ -306,6 +306,15 @@ String OS::get_system_dir(SystemDir p_dir) const {
 	return ".";
 }
 
+String OS::get_safe_application_name() const {
+	String an = Globals::get_singleton()->get("application/name");
+	Vector<String> invalid_char = String("\\ / : * ? \" < > |").split(" ");
+	for (int i=0;i<invalid_char.size();i++) {
+		an = an.replace(invalid_char[i],"-");
```

Comment:
@akien-mga it could be better to replace it to similar possible characters.
``The Test: Enclosure of <C:\Hacks>!?`` to ``The Test; Enclosure of [C;-Hacks]!!``
what do you think?